Dockerfile

```diff
@@ -1,13 +1,10 @@
-FROM continuumio/anaconda
+FROM jupyter/scipy-notebook
 
-# enable utf-8 locale
-ENV PYTHONUNBUFFERED 1
+COPY environment.yml ./
+USER root
+RUN chown jovyan:users environment.yml
+USER jovyan
```

Comment:
Is there a way to make this work for any user?

Reply:
The `jovyan` user is a user in the Docker image, not on the host (i.e. a `jovyan` user doesn't need to exist on the host).  

There is only a consideration for convenience regarding whether the UID and GID of the `jovyan` user in the Docker container matches the UID and GID of the user on the host.  When they match, files created in the container under volume mounts will be owned by the host (since the `UID:GID` property is the same).  

In the case they differ, one can set the `NB_UID` and `NB_GID` environment variables to adjust the UID and GID of the `jovyan` at runtime ([reference](https://jupyter-docker-stacks.readthedocs.io/en/latest/using/common.html)).  The only issue to resolve is that when these environment variables are set *and* we extend the image by adding our own conda environment, Jupyter fails to launch due to file permission issues.  I suspect a `chown` is failing on the new files we had to the image